pkg/didcomm/protocol/exchange/exchange.go

```diff
@@ -123,10 +138,10 @@ func encodedExchangeInvitation(inviteMessage *Invitation) (string, error) {
 	return base64.URLEncoding.EncodeToString(invitationJSON), nil
 }
 
-func marshalAndSend(data interface{}, errorMsg, destination string, transport transport.OutboundTransport) (string, error) {
+func (p *Protocol) marshalAndSend(data interface{}, errorMsg, destination string) (string, error) {
 	jsonString, err := json.Marshal(data)
 	if err != nil {
 		return "", errors.Errorf("%s : %w", errorMsg, err)
 	}
-	return transport.Send(string(jsonString), destination)
+	return p.prov.Send(string(jsonString), destination)
```

Comment:
it makes more sense to have p.outTransport.Send() instead of p.prov.Send() since the struct acting on Send() is a transport, not a provider

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/a31807a201229cd5ab417413af54914995b98b87..a26c159c381cbbb380ef97f616c8f9b9bddd476a#diff-4d3ca323f9b237cfa9940c12efa9b615R148).